src/main/java/com/revature/controllers/PurchaseController.java

```diff
@@ -0,0 +1,64 @@
+package com.revature.controllers;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.servlet.http.HttpSession;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.ResourceAccessException;
+
+import com.revature.annotations.Authorized;
+import com.revature.dtos.PurchaseRequest;
+import com.revature.models.Purchase;
+import com.revature.models.User;
+import com.revature.services.PurchaseService;
+
+@RestController
+@RequestMapping("api/purchases")
+@CrossOrigin(origins = {"http://localhost:4200", "http://localhost:3000"}, allowCredentials = "true")
```

Comment:
Change this to `@CrossOrigin(origins="*", allowedHeaders="*")` to allow frontend in prod environment to reach this controller without throwing a CORS error